protodesc/protodesc.go

```diff
@@ -172,7 +173,16 @@ func parseServiceMethod(svcAndMethod string) (string, string, error) {
 		split := strings.Split(svcAndMethod, "/")
 		return split[0], split[1], nil
 	default:
-		return "", "", newInvalidMethodNameError(svcAndMethod)
+		split := strings.Split(svcAndMethod, "/")
+		l := len(split)
+		svcSuffix, method, err := parseServiceMethod(strings.Join(split[l-2:], "/"))
+
+		if err != nil {
+			return "", "", newInvalidMethodNameError(svcAndMethod)
+		}
+
+		svcPrefix := strings.Join(split[:l-2], "/")
+			return svcPrefix + "/" + svcSuffix, method, nil
```

Comment:
This a great start.

One issue is that `ghz` looks up and verifies that the service exists in the proto file with the service definition via the method `findServiceSymbol()`:

```
func getMethodDesc(call string, files map[string]*desc.FileDescriptor) (*desc.MethodDescriptor, error) {
	svc, mth, err := parseServiceMethod(call)
	if err != nil {
		return nil, err
	}

	dsc, err := findServiceSymbol(files, svc)

```

If `ghz` is passed `a/prefix/to.a.Service/Method` as the value for `--call` command line parameter, 
then findServiceSymbol() will look for `a/prefix/to.a.Service` in the proto file and it won't be able to find `a/prefix/to.a.Service`.

For instance, if we pass `grpc/tensorflow-serving/dev-tensorboard-crossshop-ranker/tensorflow.serving.PredictionService/Predict` as the parameter for the `--call` command line argument, then the service parsed out will be grpc/tensorflow-serving/dev-tensorboard-crossshop-ranker/tensorflow.serving.PredictionService, but it has only PredictionService defined.

Passing the route separate from the Service might be a little complicated.